### DISCUSSION TITLE
Need Assistance with Flutter Issue: User Model's fromJson Method Not Executing

### DISCUSSION THREAD

**Discussion by msassa** [rrousselGit/freezed#934] [General]
Hey Flutter community,

I'm currently working on a Flutter app with a team of developers, and I've encountered a perplexing issue that I could use some help with. After logging in to the app, the fromJson method in my User Model, which is responsible for parsing JSON data, is not executing as expected.

The problem is that without the fromJson method being called, the app lacks the necessary user data, leading to unsuccessful login attempts. Strangely, there are no error messages or warnings displayed, and even adding a simple print statement within the fromJson method doesn't produce any output in the debug console. It seems that the app is not entering or executing the fromJson method at all.

The puzzling part is that this issue is specific to my machine. Other team members running the same code on their machines do not encounter this problem, which leads me to believe it might be a local configuration issue.

Here's what I've already investigated and ruled out:

Code errors: I've carefully reviewed the implementation of the fromJson method and verified that it's correctly defined and configured to handle the login data. The JSON field names match the model's properties.

Logic issues: I've double-checked the code flow to ensure that the fromJson method is called at the appropriate place after the login process. There don't appear to be any logical errors that would prevent its execution.

Network and service: I've confirmed that my machine has a stable internet connection and that the login service is functioning properly. No error messages related to communication with the login server are displayed.

Since the issue is isolated to my machine, I suspect that there might be a local configuration problem. I would greatly appreciate any suggestions or insights from the community regarding what could be causing this issue or how to further investigate it.

Here are some additional details that might help:

Operating System: Mac Monterey 12.5
Flutter SDK version: 3.7.12
Dart SDK version: 2.19.6
Editor: VSCode

Please let me know if you require any additional information.

Best regards, Mauro.

**Comment by rrousselGit** [comment]
If your `fromJson` is not executed, then it's a logic issue.

Something on your app decides not to call the `fromJson`. Or maybe it's a race condition, and the logic to call fromJson is executed _after_ the app tries to login.
It could be many things.

**Comment by msassa** [comment]
Thanks for the response.

I don't think it's a logic problem, since in all the other computers from the other developers, it works fine.
Even, running the app from vscode does not work, but running it from the terminal, with flutter run --no-sound-null-safety, sometimes it works, sometimes not.
Running it from XCode, it doesn't work either ...
The code is very simple:

```

          final token = agro24storage.accessToken;
          final userProfileService = await agro24service.getProfile(token);  <-- call the api

          final userProfileData = UserModel.fromJson(userProfileService['data']['user']); <-- parse the response, this line is never executed

```

I think is something in my environment, but I can't see what !

**Reply by rrousselGit** [reply]
> final token = agro24storage.accessToken;
>           final userProfileService = await agro24service.getProfile(token);  <-- call the api
> 
>           final userProfileData = UserModel.fromJson(userProfileService['data']['user']); <-- parse the response, this line is never executed

It could be that `getProfile` throws on your computer, and fromJson is therefore never reached

**Comment by msassa** [comment]
nop, the call works just fine, even if I print the result of that: userProfileService, the result is ok. Is a json response from the same api that the other developers use.

**Comment by konradstrg** [comment]
I have the same issue, how did you solve it?
